addons/account/models/account_tax.py

```diff
@@ -367,7 +367,7 @@ def flatten_taxes_hierarchy(self, create_map=False):
         # mapping each child tax to its parent group
         all_taxes = self.env['account.tax']
         groups_map = {}
-        for tax in self.sorted(key=lambda r: r.sequence):
+        for tax in self.sorted(key=lambda r: (r.sequence, r._origin.id)):
```

Comment:
If you want exactly the same ordering as the table, then:
```suggestion
        for tax in self.sorted():
```

Reply:
After changing sorted() to no arguments, this test fails
https://github.com/odoo/odoo/blob/a566d41427475c82e411fe4fb5225e59cb00456c/addons/account/tests/test_account_move_in_invoice.py#L758-L849
I can't figure out why. 
Could you check, please?
traceback
```
AssertionError: The records and expected_values do not match.

==== Differences at index 0 ====
--- 

+++ 

@@ -1,5 +1,5 @@

-price_unit:1152.0
-price_subtotal:960.0
-price_total:1411.2
-amount_currency:960.0
-debit:960.0
+price_unit:960.0
+price_subtotal:800.0
+price_total:1176.0
+amount_currency:800.0
+debit:800.0

==== Differences at index 7 ====
--- 

+++ 

@@ -1,5 +1,5 @@

-price_unit:-1544.0
-price_subtotal:-1544.0
-price_total:-1544.0
-amount_currency:-1544.0
-credit:1544.0
+price_unit:-1384.0
+price_subtotal:-1384.0
+price_total:-1384.0
+amount_currency:-1384.0
+credit:1384.0
```